server/paymentProviders/stripe/webhook.ts

```diff
@@ -194,26 +194,24 @@ export const paymentIntentSucceeded = async (event: Stripe.Event) => {
   await createOrUpdateOrderStripePaymentMethod(order, stripeAccount, paymentIntent);
 
   const transaction = await createChargeTransactions(charge, { order });
+  await Promise.all([
+    order.update({
+      status: !order.SubscriptionId ? OrderStatuses.PAID : OrderStatuses.ACTIVE,
+      processedAt: new Date(),
+      data: {
+        ...omit(order.data, 'paymentIntent'),
+        previousPaymentIntents: [...(order.data.previousPaymentIntents ?? []), paymentIntent],
+      },
+    }),
+    order.getOrCreateMembers(),
```

Comment:
One thing I don't like about this new approach is that any failure of `getOrCreateMembers` will prevent the creation of the subscription. Some error handling on this one could be nice.